.github/workflows/ci.yml

```diff
@@ -37,23 +50,77 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Start LocalStack
-        uses: ./
+        uses: jenseng/dynamic-uses@v1
         with:
-          image-tag: 'latest'
-          install-awslocal: 'true'
-          use-pro: 'true'
+          uses: LocalStack/setup-localstack@${{ env.action-version }}
+          with: |-
+            {
+              "image-tag": "latest",
+              "install-awslocal": "true",
+              "configuration": "DEBUG=1",
+              "use-pro": "true",
+            }
         env:
           LOCALSTACK_API_KEY: ${{ secrets.LOCALSTACK_API_KEY }}
+          action-version: ${{ fromJSON(format('["{0}","{1}"]', github.ref_name, github.head_ref))[github.event_name == 'pull_request'] }}
 
       - name: Run AWS commands
         run: |
           awslocal s3 mb s3://test
           awslocal sqs create-queue --queue-name test-queue
 
       - name: Save the Cloud Pod 
-        uses: LocalStack/setup-localstack/cloud-pods@main
+        uses: jenseng/dynamic-uses@v1
         with:
-          name: cloud-pods-test
-          action: save
+          uses: LocalStack/setup-localstack@${{ env.action-version }}
+          with: |-
+            {
+              "state-name": "cloud-pods-test",
+              "state-action": "save",
+              "skip-startup": "true",
+            }
```

Comment:
This is mostly a comment on usability:
It feels a bit weird having to supply `skip-startup` to the action, as all I want to do is save a pod - so what's with the extra arguments? (I do get where they are coming from, but I'm mostly commenting as a clueless consumer)

Given that there are clear separate use cases, I'm starting to think that batching all these actions together under an action called `setup-localstack` is not the best way.
To me it seems like there are at least two separate concerns: setting up localstack, and interacting with state.
This makes me think that ideally we'd split this up into multiple actions, `setup-localstack`, `cloud-pods`, ...

Maybe something we can tackle in a future iteration.

Reply:
The separations are existing now if you use the sub modules directly ie `setup-localstack@startup`.

The main "proxy" action is existing only to combine these items into a typical workflow, ie:
installing tools => starting up ephemeral instance => deploying infrastructure => creating PR comment => save pod => stop instance

Hence the above exist the `skip-` switches, in case you don't need certain parts of this flow.  However it's a good point to allow running save without defining `skip-startup` as you don't necessarily want to save an empty local or ephemeral instance anyways.